Log `OrderId` next to message whenever possible

luckysori opened the pull request:
It's not always available (not all messages are associated with and `OrderId`), which is why we have to use the `tracing` syntax
`?order_id`.

The context of when a message was sent or received, associated with an `OrderId`, should be useful for debugging.

klochowicz reviewed (approved):
cool, didn't know about the ?syntax 👍

da-kami commented on daemon/src/wire.rs:
```diff
@@ -159,6 +159,20 @@ impl TakerToMaker {
+            | Protocol { order_id, .. }
+            | RolloverProtocol { order_id, .. }
+            | Settlement { order_id, .. } => Some(*order_id),
+            _ => None,
```
🛠 Please don't use `_` here - if we add a new message with `order_id` we would ignore the `order_id`, if you mention all choices we will notice when adding the new message to the match.

da-kami commented on daemon/src/wire.rs:
```diff
@@ -282,6 +296,22 @@ impl MakerToTaker {
+            | ConfirmRollover { order_id, .. }
+            | RejectRollover(order_id)
+            | Settlement { order_id, .. } => Some(*order_id),
+            _ => None,
```
🛠 Same here

da-kami reviewed (approved):
ack!
But please be explicit in the match and don't use `_` :)

luckysori committed:
Provide OrderId context when sending/receiving some messages

The context of when a message was sent or received, associated with an
`OrderId`, should be useful for debugging.
